src/Caching/Hybrid/src/Internal/DefaultHybridCache.StampedeStateT.cs

```diff
@@ -0,0 +1,268 @@
+            : base(cache, key, canBeCanceled)
+        {
+            _result = new(TaskCreationOptions.RunContinuationsAsynchronously);
+        }
+
+        public override Type Type => typeof(T);
+
+        public StampedeState(DefaultHybridCache cache, in StampedeKey key, CancellationToken token)
+            : base(cache, key, token) { } // no TCS in this case - this is for SetValue only
+
+        public void QueueUserWorkItem(in TState state, Func<TState, CancellationToken, ValueTask<T>> underlying, HybridCacheEntryOptions? options)
+        {
+            Debug.Assert(_underlying is null);
+            Debug.Assert(underlying is not null);
+
+            // initialize the callback state
+            _state = state;
+            _underlying = underlying;
+            _options = options;
+
+#if NETCOREAPP3_0_OR_GREATER
+            ThreadPool.UnsafeQueueUserWorkItem(this, false);
+#else
+            ThreadPool.UnsafeQueueUserWorkItem(SharedWaitCallback, this);
+#endif
+        }
+
+        public Task ExecuteDirectAsync(in TState state, Func<TState, CancellationToken, ValueTask<T>> underlying, HybridCacheEntryOptions? options)
+        {
+            Debug.Assert(_underlying is null);
+            Debug.Assert(underlying is not null);
+
+            // initialize the callback state
+            _state = state;
+            _underlying = underlying;
+            _options = options;
+
+            return BackgroundFetchAsync();
+        }
+
+        public override void Execute() => _ = BackgroundFetchAsync();
+
+        private async Task BackgroundFetchAsync()
+        {
+            try
+            {
+                // read from L2 if appropriate
+                if ((Key.Flags & HybridCacheEntryFlags.DisableDistributedCacheRead) == 0)
+                {
+                    var result = await Cache.GetFromL2Async(Key.Key, SharedToken).ConfigureAwait(false);
+
+                    if (result.Array is not null)
+                    {
+                        SetResultAndRecycleIfAppropriate(ref result);
+                        return;
+                    }
+                }
+
+                // nothing from L2; invoke the underlying data store
+                if ((Key.Flags & HybridCacheEntryFlags.DisableUnderlyingData) == 0)
+                {
+                    var cacheItem = SetResult(await _underlying!(_state!, SharedToken).ConfigureAwait(false));
+
+                    // note that at this point we've already released most or all of the waiting callers; everything
+                    // else here is background
+
+                    // write to L2 if appropriate
+                    if ((Key.Flags & HybridCacheEntryFlags.DisableDistributedCacheWrite) == 0)
+                    {
+                        if (cacheItem.TryReserveBuffer(out var buffer))
+                        {
+                            // mutable; we've already serialized it for the shared cache item
```

Comment:
Nit: you used a colon below.

Reply:
e5a080fb6459d57472acb8c0186b26f6b451378b